Binary files SDL-1.2.14/Borland.zip and sdl-old/Borland.zip differ
Binary files SDL-1.2.14/CWprojects.sea.bin and sdl-old/CWprojects.sea.bin differ
diff -r -u -N -x .svn -x acinclude -x build-scripts -x configure.in -x docs -x 'Makefile*' -x test -x SDL.spec -x 'SDL_android*' -x Android.mk SDL-1.2.14/include/SDL_config_android.h sdl-old/include/SDL_config_android.h
--- SDL-1.2.14/include/SDL_config_android.h	1970-01-01 01:00:00.000000000 +0100
+++ sdl-old/include/SDL_config_android.h	2009-12-11 03:46:44.000000000 +0100
@@ -0,0 +1,53 @@
+/*
+    SDL - Simple DirectMedia Layer
+    Copyright (C) 1997-2009 Sam Lantinga
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+    Sam Lantinga
+    slouken@libsdl.org
+*/
+
+#ifndef _SDL_config_minimal_h
+#define _SDL_config_minimal_h
+
+#include "SDL_platform.h"
+
+/* This is the minimal configuration that can be used to build SDL */
+
+#include <stdarg.h>
+#include <stdint.h>
+
+
+#define SDL_AUDIO_DRIVER_ANDROID	1
+
+#define SDL_CDROM_DISABLED	1
+
+#define SDL_JOYSTICK_DISABLED	1
+
+#define SDL_LOADSO_DISABLED	1
+
+#define SDL_THREAD_PTHREAD	1
+#define SDL_THREAD_PTHREAD_RECURSIVE_MUTEX	1
+
+#define SDL_TIMER_UNIX	1
+
+#define SDL_VIDEO_DRIVER_ANDROID	1
+
+#define HAVE_STDIO_H	1
+
+#define NO_SDL_GLEXT	1
+
+#endif /* _SDL_config_minimal_h */
diff -r -u -N -x .svn -x acinclude -x build-scripts -x configure.in -x docs -x 'Makefile*' -x test -x SDL.spec -x 'SDL_android*' -x Android.mk SDL-1.2.14/include/SDL_config.h sdl-old/include/SDL_config.h
--- SDL-1.2.14/include/SDL_config.h	1970-01-01 01:00:00.000000000 +0100
+++ sdl-old/include/SDL_config.h	2009-12-11 03:46:44.000000000 +0100
@@ -0,0 +1,47 @@
+/*
+    SDL - Simple DirectMedia Layer
+    Copyright (C) 1997-2009 Sam Lantinga
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+    Sam Lantinga
+    slouken@libsdl.org
+*/
+
+#ifndef _SDL_config_h
+#define _SDL_config_h
+
+#include "SDL_platform.h"
+
+/* Add any platform that doesn't build using the configure system */
+#if defined(__DREAMCAST__)
+#include "SDL_config_dreamcast.h"
+#elif defined(__MACOS__)
+#include "SDL_config_macos.h"
+#elif defined(__MACOSX__)
+#include "SDL_config_macosx.h"
+#elif defined(__SYMBIAN32__)
+#include "SDL_config_symbian.h"  /* must be before win32! */
+#elif defined(__WIN32__)
+#include "SDL_config_win32.h"
+#elif defined(__OS2__)
+#include "SDL_config_os2.h"
+#elif defined(ANDROID)
+#include "SDL_config_android.h"
+#else
+#include "SDL_config_minimal.h"
+#endif /* platform config */
+
+#endif /* _SDL_config_h */
diff -r -u -N -x .svn -x acinclude -x build-scripts -x configure.in -x docs -x 'Makefile*' -x test -x SDL.spec -x 'SDL_android*' -x Android.mk SDL-1.2.14/include/SDL_config_minimal.h sdl-old/include/SDL_config_minimal.h
--- SDL-1.2.14/include/SDL_config_minimal.h	2010-01-18 10:33:32.000000000 +0100
+++ sdl-old/include/SDL_config_minimal.h	2009-12-11 03:46:44.000000000 +0100
@@ -28,7 +28,9 @@
 /* This is the minimal configuration that can be used to build SDL */
 
 #include <stdarg.h>
+#include <stdint.h>
 
+/*
 typedef signed char int8_t;
 typedef unsigned char uint8_t;
 typedef signed short int16_t;
@@ -37,26 +39,23 @@
 typedef unsigned int uint32_t;
 typedef unsigned int size_t;
 typedef unsigned long uintptr_t;
+*/
 
-/* Enable the dummy audio driver (src/audio/dummy/\*.c) */
 #define SDL_AUDIO_DRIVER_DUMMY	1
 
-/* Enable the stub cdrom driver (src/cdrom/dummy/\*.c) */
 #define SDL_CDROM_DISABLED	1
 
-/* Enable the stub joystick driver (src/joystick/dummy/\*.c) */
 #define SDL_JOYSTICK_DISABLED	1
 
-/* Enable the stub shared object loader (src/loadso/dummy/\*.c) */
 #define SDL_LOADSO_DISABLED	1
 
-/* Enable the stub thread support (src/thread/generic/\*.c) */
-#define SDL_THREADS_DISABLED	1
+#define SDL_THREAD_PTHREAD	1
+#define SDL_THREAD_PTHREAD_RECURSIVE_MUTEX	1
 
-/* Enable the stub timer support (src/timer/dummy/\*.c) */
-#define SDL_TIMERS_DISABLED	1
+#define SDL_TIMER_UNIX	1
 
-/* Enable the dummy video driver (src/video/dummy/\*.c) */
-#define SDL_VIDEO_DRIVER_DUMMY	1
+#define SDL_VIDEO_DRIVER_ANDROID	1
+
+#define HAVE_STDIO_H	1
 
 #endif /* _SDL_config_minimal_h */
diff -r -u -N -x .svn -x acinclude -x build-scripts -x configure.in -x docs -x 'Makefile*' -x test -x SDL.spec -x 'SDL_android*' -x Android.mk SDL-1.2.14/include/SDL_opengl.h sdl-old/include/SDL_opengl.h
--- SDL-1.2.14/include/SDL_opengl.h	2010-01-18 10:33:32.000000000 +0100
+++ sdl-old/include/SDL_opengl.h	2009-12-11 03:46:44.000000000 +0100
@@ -42,6 +42,9 @@
 #elif defined(__MACOS__)
 #include <gl.h>		/* Header File For The OpenGL Library */
 #include <glu.h>	/* Header File For The GLU Library */
+#elif defined(ANDROID)
+#include <GLES/gl.h>	/* Header File For The OpenGL Library */
+#include <GLES/glext.h>	/* No GLU on Android */
 #else
 #include <GL/gl.h>	/* Header File For The OpenGL Library */
 #include <GL/glu.h>	/* Header File For The GLU Library */
Binary files SDL-1.2.14/MPWmake.sea.bin and sdl-old/MPWmake.sea.bin differ
diff -r -u -N -x .svn -x acinclude -x build-scripts -x configure.in -x docs -x 'Makefile*' -x test -x SDL.spec -x 'SDL_android*' -x Android.mk SDL-1.2.14/src/audio/dma/SDL_dmaaudio.c sdl-old/src/audio/dma/SDL_dmaaudio.c
--- SDL-1.2.14/src/audio/dma/SDL_dmaaudio.c	2010-01-18 10:31:46.000000000 +0100
+++ sdl-old/src/audio/dma/SDL_dmaaudio.c	2009-12-11 03:46:44.000000000 +0100
@@ -40,7 +40,7 @@
 #include <soundcard.h>
 #else
 /* This is recommended by OSS */
-#include <sys/soundcard.h>
+#include <linux/soundcard.h>
 #endif
 
 #ifndef MAP_FAILED
diff -r -u -N -x .svn -x acinclude -x build-scripts -x configure.in -x docs -x 'Makefile*' -x test -x SDL.spec -x 'SDL_android*' -x Android.mk SDL-1.2.14/src/audio/dsp/SDL_dspaudio.c sdl-old/src/audio/dsp/SDL_dspaudio.c
--- SDL-1.2.14/src/audio/dsp/SDL_dspaudio.c	2010-01-18 10:31:52.000000000 +0100
+++ sdl-old/src/audio/dsp/SDL_dspaudio.c	2009-12-11 03:46:44.000000000 +0100
@@ -41,7 +41,7 @@
 #include <soundcard.h>
 #else
 /* This is recommended by OSS */
-#include <sys/soundcard.h>
+#include <linux/soundcard.h>
 #endif
 
 #include "SDL_timer.h"
diff -r -u -N -x .svn -x acinclude -x build-scripts -x configure.in -x docs -x 'Makefile*' -x test -x SDL.spec -x 'SDL_android*' -x Android.mk SDL-1.2.14/src/audio/SDL_audio.c sdl-old/src/audio/SDL_audio.c
--- SDL-1.2.14/src/audio/SDL_audio.c	2010-01-18 10:31:56.000000000 +0100
+++ sdl-old/src/audio/SDL_audio.c	2009-12-11 03:46:44.000000000 +0100
@@ -113,6 +113,9 @@
 #if SDL_AUDIO_DRIVER_EPOCAUDIO
 	&EPOCAudio_bootstrap,
 #endif
+#if SDL_AUDIO_DRIVER_ANDROID
+	&ANDROIDAUD_bootstrap,
+#endif
 	NULL
 };
 SDL_AudioDevice *current_audio = NULL;
diff -r -u -N -x .svn -x acinclude -x build-scripts -x configure.in -x docs -x 'Makefile*' -x test -x SDL.spec -x 'SDL_android*' -x Android.mk SDL-1.2.14/src/audio/SDL_sysaudio.h sdl-old/src/audio/SDL_sysaudio.h
--- SDL-1.2.14/src/audio/SDL_sysaudio.h	2010-01-18 10:31:56.000000000 +0100
+++ sdl-old/src/audio/SDL_sysaudio.h	2009-12-11 03:46:44.000000000 +0100
@@ -177,6 +177,9 @@
 #if SDL_AUDIO_DRIVER_EPOCAUDIO
 extern AudioBootStrap EPOCAudio_bootstrap; 
 #endif
+#if SDL_AUDIO_DRIVER_ANDROID
+extern AudioBootStrap ANDROIDAUD_bootstrap;
+#endif
 
 /* This is the current audio device */
 extern SDL_AudioDevice *current_audio;
diff -r -u -N -x .svn -x acinclude -x build-scripts -x configure.in -x docs -x 'Makefile*' -x test -x SDL.spec -x 'SDL_android*' -x Android.mk SDL-1.2.14/src/video/fbcon/SDL_fbvideo.c sdl-old/src/video/fbcon/SDL_fbvideo.c
--- SDL-1.2.14/src/video/fbcon/SDL_fbvideo.c	2010-01-18 10:32:28.000000000 +0100
+++ sdl-old/src/video/fbcon/SDL_fbvideo.c	2009-12-11 03:46:44.000000000 +0100
@@ -200,7 +200,7 @@
 	/* Added check for /fb/0 (devfs) */
 	/* but - use environment variable first... if it fails, still check defaults */
 	int idx = 0;
-	const char *SDL_fbdevs[4] = { NULL, "/dev/fb0", "/dev/fb/0", NULL };
+	const char *SDL_fbdevs[4] = { NULL, "/dev/graphics/fb0", "/dev/fb/0", NULL };
 
 	SDL_fbdevs[0] = SDL_getenv("SDL_FBDEV");
 	if( !SDL_fbdevs[0] )
@@ -509,7 +509,7 @@
 	/* Initialize the library */
 	SDL_fbdev = SDL_getenv("SDL_FBDEV");
 	if ( SDL_fbdev == NULL ) {
-		SDL_fbdev = "/dev/fb0";
+		SDL_fbdev = "/dev/graphics/fb0";
 	}
 	console_fd = open(SDL_fbdev, O_RDWR, 0);
 	if ( console_fd < 0 ) {
diff -r -u -N -x .svn -x acinclude -x build-scripts -x configure.in -x docs -x 'Makefile*' -x test -x SDL.spec -x 'SDL_android*' -x Android.mk SDL-1.2.14/src/video/SDL_sysvideo.h sdl-old/src/video/SDL_sysvideo.h
--- SDL-1.2.14/src/video/SDL_sysvideo.h	2010-01-18 10:32:46.000000000 +0100
+++ sdl-old/src/video/SDL_sysvideo.h	2009-12-11 03:46:44.000000000 +0100
@@ -410,6 +410,9 @@
 #if SDL_VIDEO_DRIVER_CACA
 extern VideoBootStrap CACA_bootstrap;
 #endif
+#if SDL_VIDEO_DRIVER_ANDROID
+extern VideoBootStrap ANDROID_bootstrap;
+#endif
 #if SDL_VIDEO_DRIVER_DUMMY
 extern VideoBootStrap DUMMY_bootstrap;
 #endif
diff -r -u -N -x .svn -x acinclude -x build-scripts -x configure.in -x docs -x 'Makefile*' -x test -x SDL.spec -x 'SDL_android*' -x Android.mk SDL-1.2.14/src/video/SDL_video.c sdl-old/src/video/SDL_video.c
--- SDL-1.2.14/src/video/SDL_video.c	2010-01-18 10:32:46.000000000 +0100
+++ sdl-old/src/video/SDL_video.c	2009-12-11 03:46:44.000000000 +0100
@@ -126,6 +126,9 @@
 #if SDL_VIDEO_DRIVER_CACA
 	&CACA_bootstrap,
 #endif
+#if SDL_VIDEO_DRIVER_ANDROID
+	&ANDROID_bootstrap,
+#endif
 #if SDL_VIDEO_DRIVER_DUMMY
 	&DUMMY_bootstrap,
 #endif
Binary files SDL-1.2.14/symbian.zip and sdl-old/symbian.zip differ
Binary files SDL-1.2.14/VisualCE.zip and sdl-old/VisualCE.zip differ
Binary files SDL-1.2.14/VisualC.zip and sdl-old/VisualC.zip differ
Binary files SDL-1.2.14/Watcom-OS2.zip and sdl-old/Watcom-OS2.zip differ
Binary files SDL-1.2.14/Watcom-Win32.zip and sdl-old/Watcom-Win32.zip differ
Binary files SDL-1.2.14/Xcode.tar.gz and sdl-old/Xcode.tar.gz differ
